Microsoft.Toolkit.Uwp/Helpers/DispatcherHelper.cs

```diff
@@ -258,12 +260,27 @@ public static Task<T> AwaitableRunAsync<T>(this CoreDispatcher dispatcher, Func<
         /// <returns>Awaitable Task</returns>
         public static Task AwaitableRunAsync(this CoreDispatcher dispatcher, Action function, CoreDispatcherPriority priority = CoreDispatcherPriority.Normal)
         {
-            return dispatcher.AwaitableRunAsync<object>(
-                () =>
+            if (function == null)
+            {
+                throw new ArgumentNullException("function can't be null!");
+            }
+
+            var taskCompletionSource = new TaskCompletionSource<object>();
```

Comment:
Here we only save one lambda and one method call. No await is saved, so I can revert to the old implementation if you like it better. It won't be much of a difference performance wise.